### ISSUE TITLE
UI silently changes price on small orders

### ISSUE THREAD

**Issue by theoreticalbts** [steemit/steemit.com#95]
Continuing discussion of https://github.com/steemit/steem/issues/234

The issue here is that if the user order is very small, the requested price may not be possible.  For example if you are buying Steem with 96 SD, getting 35 Steem would give you a price of $2.7428 while getting 36 Steem would give you a price of $2.8235, and no price between these two points is possible (for example in the screenshot the user requested a price of $2.76).

Basically I think the price field needs to be updated (and perhaps a warning given) if the real price (as determined by SD field / STEEM field) is more than $0.01 different than the Price field.  Or it might be $0.005 SD difference depending on how the rounding works out -- whoever fixes this should carefully consider how the rounding semantics should work.

**Comment by roadscape** [comment]
> if you are buying Steem with 96 SD, getting 35 Steem would give you a price of $2.7428 while getting 36 Steem would give you a price of $2.8235, and no price between these two points is possible

You must be referring to **satoshi** amounts here.. i.e. amounts of $0.096 and 0.035 STEEM.

Very small orders are susceptible to rounding issues in the UI. Do note that the confirmation dialog shows the effective rate. Perhaps we should highlight it in red when it differs from what is typed in the form.

> UI silently changes price on small orders

The UI doesn't "change" any price, that's the one field it won't auto-calculate. The user sets the price (by keying one in or using the price of an existing order), and the form should auto-calculate the nearest *valid* STEEM or SD amount, as the SD or STEEM amount changes, respectively.

----

Buyer: I'm buying 0.01 apples at $1.10 each.
Seller: That will be $0.02
Buyer: But that's $2/apple

Possible solutions:

1) Do not allow tiny orders
2) Highlight the price field when the effective price is off by more than a certain percent
3) Highlight effective price in confirmation dialog when it's different from the form

**Comment by roadscape** [comment]
For reference, from the original error report: https://cloud.githubusercontent.com/assets/5168676/17714358/67ac8232-63cd-11e6-938a-5029304dfa79.png

 - Entered price: 2.76
 - Effective price: 2.742857 (0.62% lower)

After adjusting the rounding algorithm, I see it rounds the other way now (as it should have been):

 - Entered price: 2.76
 - Effective price: 2.771428 (0.41% higher)

-----

I experimented with a simple indicator if the effective price is more than 1% different from the entered price. However, this is pretty hard to trigger. As far as I can tell, as long as the order total is above $0.10 or so it is hard to face a rounding error of more than 1%.

Given that the original error report uses orders of **less than $0.10** and *still* results in a error of **less than 1%** I'd say this is nearly a non-issue.

However, I've written some code which detects these cases (price is different from effective price by more than 1%) and de-emphasizes the price on the form: opacity gets set to 25%, making it very light, signaling the user to take a closer look at the price on the confirm order form. Anything more complex wouldn't make sense for a 'rounding issue' on dust orders.

![image](https://cloud.githubusercontent.com/assets/5168676/17739849/7d41e6ae-6464-11e6-9d08-44df631130cd.png)
